tests/mockserver_tests/mock_server_test_base.py

```diff
@@ -228,3 +229,109 @@ def database(self) -> Database:
+            CreateSessionRequest,
+        )
+
+        mux_enabled = is_multiplexed_enabled(transaction_type)
+        idx = 0
+        # Skip all leading BatchCreateSessionsRequest (for retries)
+        if allow_multiple_batch_create:
+            while idx < len(requests) and isinstance(
+                requests[idx], BatchCreateSessionsRequest
+            ):
+                idx += 1
+            # For multiplexed, optionally skip a CreateSessionRequest
+            if (
+                mux_enabled
+                and idx < len(requests)
+                and isinstance(requests[idx], CreateSessionRequest)
+            ):
+                idx += 1
+        else:
+            if mux_enabled:
+                self.assertTrue(
+                    isinstance(requests[idx], BatchCreateSessionsRequest),
+                    f"Expected BatchCreateSessionsRequest at index {idx}, got {type(requests[idx])}",
+                )
+                idx += 1
+                self.assertTrue(
+                    isinstance(requests[idx], CreateSessionRequest),
+                    f"Expected CreateSessionRequest at index {idx}, got {type(requests[idx])}",
+                )
+                idx += 1
+            else:
+                self.assertTrue(
+                    isinstance(requests[idx], BatchCreateSessionsRequest),
+                    f"Expected BatchCreateSessionsRequest at index {idx}, got {type(requests[idx])}",
+                )
+                idx += 1
+        # Check the rest of the expected request types
+        for expected_type in expected_types:
+            self.assertTrue(
+                isinstance(requests[idx], expected_type),
+                f"Expected {expected_type} at index {idx}, got {type(requests[idx])}",
+            )
+            idx += 1
+        self.assertEqual(
+            idx, len(requests), f"Expected {idx} requests, got {len(requests)}"
+        )
+
+    def adjust_request_id_sequence(self, expected_segments, requests, transaction_type):
+        """Adjust expected request ID sequence numbers based on actual session creation requests.
+
+        Args:
+            expected_segments: List of expected (method, (sequence_numbers)) tuples
+            requests: List of actual requests from spanner_service.requests
+            transaction_type: TransactionType enum value to check multiplexed session status
+
+        Returns:
+            List of adjusted expected segments with corrected sequence numbers
+        """
+        from google.cloud.spanner_v1 import (
+            BatchCreateSessionsRequest,
+            CreateSessionRequest,
+            ExecuteSqlRequest,
+            BeginTransactionRequest,
+        )
+
+        # Count session creation requests that come before the first non-session request
+        session_requests_before = 0
+        for req in requests:
+            if isinstance(req, (BatchCreateSessionsRequest, CreateSessionRequest)):
+                session_requests_before += 1
```

Comment:
Will BatchCreateSessionsRequest still happen, if Mux sessions are enabled. If No, then we should only check for CreateSessionRequest here?

Reply:
It will still happen, because on can disable mux for individual operation like read-only, read-write or partition ops, at that time client will fallback to use regular pools which are initialized via these batch create calls